File: ts-reference/ui/theme.ts
import type { Role } from "../store";

export const theme = {
  title: "tuichat",
  colors: {
    user: "#7dd3fc",
    agent: "#a78bfa",
    system: "#6b7280",
    timestamp: "#4b5563",
    border: "#374151",
    suggestionBg: "#1f2937",
    suggestionSelectedBg: "#374151",
    typing: "#f59e0b",
    attachment: "#34d399",
    custom: "#f472b6",
    input: "#e5e7eb",
    prompt: "#7dd3fc",
  },
  prefix: {
    user: "you",
    agent: "agent",
    system: "sys",
  } satisfies Record<Role, string>,
};

export const formatTime = (d: Date): string => {
  const hh = d.getHours().toString().padStart(2, "0");
  const mm = d.getMinutes().toString().padStart(2, "0");
  const ss = d.getSeconds().toString().padStart(2, "0");
  return `${hh}:${mm}:${ss}`;
};

export const roleColor = (role: Role): string => theme.colors[role];

export const formatBytes = (bytes: number): string => {
  if (bytes < 1024) return `${bytes}B`;
  if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)}kB`;
  return `${(bytes / 1024 / 1024).toFixed(1)}MB`;
};

File: internal/ui/theme.go
package ui

import "github.com/charmbracelet/lipgloss"

type Theme struct {
	Title                string
	UserColor            lipgloss.Color
	AgentColor           lipgloss.Color
	SystemColor          lipgloss.Color
	TimestampColor       lipgloss.Color
	BorderColor          lipgloss.Color
	SuggestionBG         lipgloss.Color
	SuggestionSelectedBG lipgloss.Color
	TypingColor          lipgloss.Color
	AttachmentColor      lipgloss.Color
	CustomColor          lipgloss.Color
	InputColor           lipgloss.Color
	PromptColor          lipgloss.Color
}

var DefaultTheme = Theme{
	Title:                "tuichat",
	UserColor:            lipgloss.Color("#7dd3fc"),
	AgentColor:           lipgloss.Color("#a78bfa"),
	SystemColor:          lipgloss.Color("#6b7280"),
	TimestampColor:       lipgloss.Color("#4b5563"),
	BorderColor:          lipgloss.Color("#374151"),
	SuggestionBG:         lipgloss.Color("#1f2937"),
	SuggestionSelectedBG: lipgloss.Color("#374151"),
	TypingColor:          lipgloss.Color("#f59e0b"),
	AttachmentColor:      lipgloss.Color("#34d399"),
	CustomColor:          lipgloss.Color("#f472b6"),
	InputColor:           lipgloss.Color("#e5e7eb"),
	PromptColor:          lipgloss.Color("#7dd3fc"),
}

func (t Theme) RolePrefix(role string) string {
	switch role {
	case "user":
		return "you  "
	case "agent":
		return "agent"
	case "system":
		return "sys  "
	}
	return "     "
}

func (t Theme) RoleColor(role string) lipgloss.Color {
	switch role {
	case "user":
		return t.UserColor
	case "agent":
		return t.AgentColor
	}
	return t.SystemColor
}
